.github/workflows/build-publish.yaml

```diff
@@ -0,0 +1,141 @@
+        with:
+          work-dir: chart
+  test-app:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Go environment
+        uses: actions/setup-go@v3.3.1
+        with:
+          go-version-file: go.mod
+          cache: true
+          cache-dependency-path: go.sum
+      - name: Setup dependencies
+        run: go mod download
+      - name: Run tests
+        run: go test ./...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - uses: actions/checkout@v3
+      - name: Build Docker image
+        uses: docker/build-push-action@v3.2.0
+        with:
+          context: .
+          push: false
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+  publish:
+    if: github.ref == 'refs/heads/master'
+    needs:
+      - test-app
+      - test-chart
+      - build
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: write
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2.1.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - uses: docker/setup-buildx-action@v2
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2.1.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Conventional Changelog Action
+        id: changelog
+        uses: TriPSs/conventional-changelog-action@v3
+        with:
+          fallback-version: 0.0.0
+          skip-commit: true
+          skip-version-file: true
+          git-push: false
+
+      - name: Update appVersion in chart
+        if: ${{ steps.changelog.outputs.skipped == 'false' }}
+        uses: mikefarah/yq@v4
+        with:
+          cmd: yq -i '.appVersion = "${{ steps.changelog.outputs.version }}"' 'chart/Chart.yaml'
+
+      - name: Commit release
+        if: ${{ steps.changelog.outputs.skipped == 'false' }}
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git add .
+          git commit -m "chore(release): ${{ steps.changelog.outputs.tag }}"
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        if: ${{ steps.changelog.outputs.skipped == 'false' }}
+        uses: docker/metadata-action@v4.1.1
+        with:
+          images: ghcr.io/${{ github.repository }}
+          tags: |
+            type=raw,value=latest
+            type=semver,pattern={{version}},value=${{ steps.changelog.outputs.tag }}
```

Comment:
This is the magic of the docker-metadata action! :D 

See [here](https://github.com/docker/metadata-action#typesemver)

TL;DR using `type=semver` you pass the semver-compliant git tag and a pattern, and it will create the appropriate docker tag for the image.

Reply:
That I understand, but my point was that `.tag` is _not_ semver-compliant, right?

Reply:
Oh, yeah, I see your point. I just went with this way since its expected to use the tag, according to the docs 🤷🏼

Reply:
But that would probably break if tag prefix was anything else than `""`, right?

Reply:
So we'll use `.version`, right?

Reply:
> But that would probably break if tag prefix was anything else than "", right?

Probably, but I honestly don't know. The docs don't say much about that. 

> So we'll use .version, right?

Sure, if that helps you sleep at night ;) The version tagged on this project will be the same for both the chart and app, `v0.0.0`, so it doesn't really make a difference